presto-iceberg/src/test/java/io/prestosql/plugin/iceberg/TestIcebergSmoke.java

```diff
@@ -587,6 +595,112 @@ private void testPredicating(Session session, FileFormat fileFormat)
+        assertQuery(session, "SELECT COUNT(*) FROM \"test_metadata_delete_simple$partitions\"", "SELECT 2");
+        dropTable(session, "test_metadata_delete_simple");
+    }
+
+    @Test
+    public void testMetadataDelete()
+    {
+        testWithAllFileFormats(this::testMetadataDeleteForFormat);
+    }
+
+    private void testMetadataDeleteForFormat(Session session, FileFormat format)
+    {
+        @Language("SQL") String createTable = "" +
+                "CREATE TABLE test_metadata_delete " +
+                "(" +
+                "  ORDER_KEY BIGINT," +
+                "  LINE_NUMBER INTEGER," +
+                "  LINE_STATUS VARCHAR" +
+                ") " +
+                "WITH (" +
+                format(" format = '%s', partitioning = ARRAY[ 'LINE_NUMBER', 'LINE_STATUS' ]", format.name()) +
+                ") ";
+
+        assertUpdate(session, createTable);
+
+        assertUpdate(session, "" +
+                        "INSERT INTO test_metadata_delete " +
+                        "SELECT orderkey, linenumber, linestatus " +
+                        "FROM tpch.tiny.lineitem",
+                "SELECT count(*) FROM lineitem");
+
+        assertQuery(session, "SELECT COUNT(*) FROM \"test_metadata_delete$partitions\"", "SELECT 14");
+
+        assertUpdate(session, "DELETE FROM test_metadata_delete WHERE LINE_STATUS='F' AND LINE_NUMBER=CAST(3 AS INTEGER)");
+        assertQuery(session, "SELECT * FROM test_metadata_delete", "SELECT orderkey, linenumber, linestatus FROM lineitem WHERE linestatus<>'F' or linenumber<>3");
+        assertQuery(session, "SELECT COUNT(*) FROM \"test_metadata_delete$partitions\"", "SELECT 13");
+
+        assertUpdate(session, "DELETE FROM test_metadata_delete WHERE LINE_STATUS='O'");
+        assertQuery(session, "SELECT COUNT(*) FROM \"test_metadata_delete$partitions\"", "SELECT 6");
+        assertQuery(session, "SELECT * FROM test_metadata_delete", "SELECT orderkey, linenumber, linestatus FROM lineitem WHERE linestatus<>'O' AND linenumber<>3");
+
+        assertQueryFails("DELETE FROM test_metadata_delete WHERE ORDER_KEY=1", "This connector only supports delete where one or more partitions are deleted entirely");
+
+        assertUpdate(session, "DROP TABLE test_metadata_delete");
+
+        assertFalse(getQueryRunner().tableExists(getSession(), "test_metadata_delete"));
+    }
+
+    @Test
+    public void testInSmallSet()
+    {
+        testWithAllFileFormats(this::testInSmallSetForFormat);
+    }
+
+    private void testInSmallSetForFormat(Session session, FileFormat format)
+    {
+        assertUpdate(session, "CREATE TABLE test_in_small_set (col1 INTEGER, col2 BIGINT)");
+        assertUpdate(session, "INSERT INTO test_in_small_set VALUES (1, 10)", 1);
+        MaterializedResult result = computeActual("SELECT col1 FROM test_in_small_set WHERE col1 IN (1)");
+        // The query below fails in h2 with the col1 IN (1) or col1 = 1.  It works fine in Iceberg.
```

Comment:
This is expected since H2 doesn't have the `test_in_small_set`. Please remove comment

Reply:
Removed along with this test, since it covers a subset of what's tested by the following test.